### DISCUSSION TITLE
"vercel dev" not working with vite - hangs indefinitely

### DISCUSSION THREAD

**Discussion by sargant** [vercel/vercel#7139] [Help] [answered]
I'm having trouble getting Vite to work with the `vercel dev` command. HTTP requests never load, and time out after 5 minutes.

Replicable environment steps:
* `npm init vite@latest vercel-test --template vanilla`
* `cd vercel-test && npm install`
* `npm install --save-dev vercel@23.1.2`
* In `package.json`, add `"start": "vercel dev"` to `scripts`
* `npm start` and point a browser at http://localhost:3000

The request remains pending for five minutes before eventually timing out.

I've tried it with `create-react-app` instead of `vite` and it works fine; and the `vite` dev command works fine by itself independently.

**Comment by sargant** [comment, marked as answer]
Seems like the problem is that the default _Development Command_ for Vite projects doesn't work out of the box.

The default is `vite` - overriding it to `vite --port $PORT` works.

**Reply by REX500** [reply]
This doesn't really help since when i run `vercel dev`, one of the 2nd arguments that get logged out are:
```
> Running Dev Command “vite --port $PORT”
> Ready! Available at http://localhost:3000
```
However, when I navigate to localhost:3000 on my browser, I get
```
GET http://localhost:3000/src/main.tsx net::ERR_ABORTED 404 (Not Found)
```
type errors... My vite app works just fine if I run it with `yarn dev` but with `vercel dev` it doesn't. Note: my app works just fine on Vercel itself...

**Reply by REX500** [reply]
I am guessing that I need to create something like `vercel.config.ts` file? I tried finding docs on this but to no avail... all of them point to servless setup but not the local vercel server itself...

**Reply by nicehatharry** [reply]
You need to set the dev command in your vercel.json

https://vercel.com/docs/projects/project-configuration#devcommand

**Comment by AlenDavid** [comment]
Hey, it seems this is still happening, now for node 18.

**Comment by martin-dimi** [comment]
Hey, I'm having the same issue. Any workarounds? We can't set another default port

**Comment by mirshko** [comment]
It seems that the suggested rewrite rule for SPA's is breaking `vercel dev` 

in vercel.json:

```json
{
      "source": "/(.*)",
      "destination": "/index.html"
}
```

After removing this rewrite rule, `vercel dev` with Vite works correctly.  It seems it is automatically performing this rewrite in dev-mode without requiring it being set? Give that a shot!

**Reply by mirshko** [reply]
@leerob Maybe you can shed some light on this?
